clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/appgateway/model/AzureAppGatewayDescription.groovy

```diff
@@ -71,8 +75,30 @@ class AzureAppGatewayDescription extends AzureResourceOpsDescription {
     description.detail = appGateway.tags?.detail ?: parsedName.detail
     description.appName = appGateway.tags?.appName ?: parsedName.app
     description.loadBalancerName = appGateway.name
-    description.cluster = appGateway.tags?.cluster
-    description.serverGroups = appGateway.tags?.serverGroups?.split(" ")
+
+    // Get current backend address pool id from the application gateway requested routing rules
+    def bapActiveRuleId = appGateway.requestRoutingRules?.first()?.backendAddressPool?.id
+    if (bapActiveRuleId && AzureUtilities.getNameFromResourceId(bapActiveRuleId) != AzureAppGatewayResourceTemplate.defaultAppGatewayBeAddrPoolName) {
+      description.trafficEnabledSG = AzureUtilities.getNameFromResourceId(bapActiveRuleId)
+      description.cluster = Names.parseName(description.trafficEnabledSG).cluster
+    } else {
+      description.trafficEnabledSG = appGateway.tags?.trafficEnabledSG
+      description.cluster = appGateway.tags?.cluster
+    }
+
+    // Each application gateway backend address pool corresponds to a server group (except the "defaul_BAP0")
+    description.serverGroups = []
+    appGateway.backendAddressPools?.each { bap ->
+      if (bap.name != AzureAppGatewayResourceTemplate.defaultAppGatewayBeAddrPoolName) description.serverGroups << bap.name
+    }
+
+    description.publicIpId = appGateway.frontendIPConfigurations
+
+    // We only support one subnet so we can just retrieve the first one
+    def subnetId = appGateway?.gatewayIPConfigurations?.first()?.subnet?.id
+    description.subnet = appGateway.tags?.subnet ?: AzureUtilities.getNameFromResourceId(subnetId)
+    // assert AzureUtilities.getNameFromResourceId(subnetId) == description.subnet
```

Comment:
It's more of a note to myself as to what the values should be to make sure that the system is consistent while debugging through the code. But it can be removed now.